### PR TITLE
5.2.0 updates

### LINKED ISSUE #70: Guide Updates for v. 5.2.0
Updates needed for new API being added to v. 5.2.0:
1. [ ] Show Manager
1. [ ] Image Manager
1. [ ] Change Registration Manager
1. [ ] Batch Sending RPCs
1. [ ] Additional HMI Types

### PR DESCRIPTION
**Pull request by NicoleYarroch**
Fixes #70

This pull request **fixes existing content** and **adds new content**.

## Summary of Changes
1. Added documentation for new API
    * Image manager
    * Show manager
    * Change registration manager
    * Batch sending RPCs
1. Added documentation for `SDLSubscribeButton`s

## Bug Fixes
* Fixed broken code placeholders in the **Text Images and Buttons** section.

## Tasks Remaining:
1. [x] Batch sending RPCs
1. [x] Proof reading

### CODE REVIEW

File: docs/Displaying Information/Text Images and Buttons/index.md [hunk 1]
```diff
@@ -46,28 +68,26 @@ show.graphic = image;
...
-      // The text has been set successfully
+      <#Image Sent Successfully#>
     }
 }];
 ```
 
 #### Swift
 ```swift
-let sdlImage = SDLImage(name: "<#Uploaded As Name", ofType: .dynamic)
+let sdlImage = SDLImage(name: "<#Uploaded As Name#>", ofType: .dynamic)
```
Comment (joeljfischer) [inline docs/Displaying Information/Text Images and Buttons/index.md]: We have an initializer with just `name` now

File: docs/Getting Started/Adapting to the Head Unit Language/index.md [hunk 2]
```diff
@@ -22,67 +24,46 @@ let headUnitHMIDisplayLanguage = sdlManager.registerResponse?.hmiDisplayLanguage
...
-2. If your app supports the head unit's current `language` and `hmiDisplayLanguage` and the app's default language is different from the head unit's current language, send a `SDLChangeRegistration` RPC with the new app name.
-
-It is important to note, that when the head unit's language is changed by the user, the head unit will send a  `OnAppInterfaceUnregistered` notification to the app. The app will then disconnect and reconnect automatically to the head unit. After reconnection, the `registerAppInterfaceResponse` will contain the new `language` and `hmiDisplayLanguage`.
+To customize the app name for the head unit's current language, implement the following steps:
+1. Set the default `language` in the `SDLLifecycleConfiguration`.
+2. Add all languages your app supports to `languagesSupported` in the `SDLLifecycleConfiguration`.
+3. Implement the `SDLManagerDelegate`'s `managerShouldUpdateLifecycleToLanguage:` method. If the head unit's language is different from the default language and is a supported language, the method will be called with the head unit's current language. Return a `SDLLifecycleConfigurationUpdate` object with the new `appName` and/or  `ttsName`.
```
Comment (joeljfischer) [inline docs/Getting Started/Adapting to the Head Unit Language/index.md]: Extra space

File: docs/Getting Started/Adapting to the Head Unit Language/index.md [hunk 3]
```diff
@@ -22,67 +24,46 @@ let headUnitHMIDisplayLanguage = sdlManager.registerResponse?.hmiDisplayLanguage
...
 
 #### Objective-C
 ```objc
-SDLChangeRegistration *changeRegistration = [[SDLChangeRegistration alloc] initWithLanguage:<#Matching language#> hmiDisplayLanguage:<#Matching language#> appName:@"<#App name for new language#>" ttsName:@[<#App TTS name for language#>] ngnMediaScreenAppName:<#NGN media screen app name#> vrSynonyms:<#VR synonyms#>];
+- (nullable SDLLifecycleConfigurationUpdate *)managerShouldUpdateLifecycleToLanguage:(SDLLanguage)language {
+    SDLLifecycleConfigurationUpdate *configurationUpdate = [[SDLLifecycleConfigurationUpdate alloc] init];
+
+    if ([language isEqualToEnum:SDLLanguageEnUs]) {
+        update.appName = <#App Name in English#>;
+    } else if ([language isEqualToString:SDLLanguageEsMx]) {
```
Comment (joeljfischer) [inline docs/Getting Started/Adapting to the Head Unit Language/index.md]: Either use `isEqualToEnum:` or String, but not both.

File: docs/Batch Sending RPCs/index.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+```swift
+sdlManager.send([<#SDLRPCRequest#>], progressHandler: { (request, response, error, percentComplete) in
+    print("Command \(request.name) sent \(response?.resultCode == .success ? "successfully" : "unsuccessfully"), percent complete \(percentComplete * 100)")
+}) { success in
+    print("All requests sent \(success ? "successfully" : "unsuccessfully")")
+}
+```
+
+### Send Sequential Requests
+Requests sent sequentially are sent in a set order. Only when a response has been received for the previously sent request, is the next request sent.
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply docs/Batch Sending RPCs/index.md]: Fixed

File: config.json [hunk 5]
```diff
@@ -46,6 +52,9 @@
         {
             "name": "Uploading Files and Graphics"
         },
+        {
+            "name": "Batch Sending RPCs"
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply config.json]: Changed.

File: docs/Displaying Information/Text Images and Buttons/index.md (hunk 1 above)
Comment (NicoleYarroch) [reply docs/Displaying Information/Text Images and Buttons/index.md]: Fixed

File: docs/Getting Started/Adapting to the Head Unit Language/index.md (hunk 3 above)
Comment (NicoleYarroch) [reply docs/Getting Started/Adapting to the Head Unit Language/index.md]: Fixed

File: docs/Getting Started/Adapting to the Head Unit Language/index.md (hunk 2 above)
Comment (NicoleYarroch) [reply docs/Getting Started/Adapting to the Head Unit Language/index.md]: Fixed

File: docs/Batch Sending RPCs/index.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+## Batch Sending RPCs
+There are two ways to send multiple requests to the head unit: concurrently and sequentially. Which method you should use depends on the type of RPCs being sent. Concurrently sent requests will not finish in any set order and should only be used when none of the requests depend on the result of another, such as when uploading group of artworks. Sequentially sent requests only send the next request in the group when a response has been received for the previously sent RPC. Requests should be sent sequentially when Core expects RPCs to be sent in a particular order like when sending the several different requests needed to create a menu.
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply docs/Batch Sending RPCs/index.md]: Fixed. I added examples to both types of batch sends

File: docs/Getting Started/Batch Sending RPCs/index.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
...
+There are two ways to send multiple requests to the head unit: concurrently and sequentially. Which method you should use depends on the type of RPCs being sent. Concurrently sent requests will finish in a random order and should only be used when none of the requests depend on the result of another, such as when uploading group of artworks. Sequentially sent requests only send the next request in the group when a response has been received for the previously sent RPC. Requests should be sent sequentially when Core expects RPCs to be sent in a particular order, like when sending the several different requests needed to create a menu.
```
Comment (joeljfischer) [inline docs/Getting Started/Batch Sending RPCs/index.md]: Concurrently sent requests won't necessarily finish randomly, but they *could* finish out of order. They are sent in sequence, but the head unit could respond out of order. So even if Core needs them sent in a certain sequence, you can use the concurrent batching, it's only when you want to know the result of a previous one or timing is more of an issue (e.g. you need Core to finish processing one before you send the next) that you need the "sequential".

**Pull request merged**